cmd/dvol-docker-plugin/dvol-docker-plugin.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+    "fmt"
+    "encoding/json"
+    "errors"
+    "log"
+    "net"
+    "net/http"
+    "os"
+    "io/ioutil"
+
+	"github.com/ClusterHQ/dvol/pkg/api"
+)
+
+const PLUGINS_DIR = "/run/docker/plugins"
+const DVOL_SOCKET = PLUGINS_DIR + "/dvol.sock"
+const VOL_DIR = "/var/lib/dvol/volumes"
+
+type ResponseImplements struct {
+    // A response to the Plugin.Activate request
+    Implements []string
+}
+
+type RequestCreate struct {
+    // A request to create a volume for Docker
+    Name string
+    Opts map[string]string
+}
+
+type RequestMount struct {
+    // A request to mount a volume for Docker
+    Name string
+}
+
+type RequestRemove struct {
+    // A request to remove a volume for Docker
+    Name string
+}
+
+type ResponseSimple struct {
+    // A response which only indicates if there was an error or not
+    Err string
+}
+
+type ResponseMount struct {
+    // A response to the VolumeDriver.Mount request
+    Mountpoint string
+    Err string
+}
+
+func main () {
+    if _, err := os.Stat(PLUGINS_DIR); err != nil {
+        if err := os.MkdirAll(PLUGINS_DIR, 0700); err != nil {
```

Comment:
I'm not sure if there's a way to do this in Go but this would look better joined by an 'and'